src/client/datascience/raw-kernel/rawSocket.ts

```diff
@@ -159,10 +160,48 @@ export class RawSocket implements IWebSocketLike, IKernelSocket, IDisposable {
 
         // Wire up all of the different channels.
         const result: IChannels = {
-            iopub: this.generateChannel(connection, 'iopub', () => new zmq.Subscriber()),
-            shell: this.generateChannel(connection, 'shell', () => new zmq.Dealer({ routingId })),
-            control: this.generateChannel(connection, 'control', () => new zmq.Dealer({ routingId })),
-            stdin: this.generateChannel(connection, 'stdin', () => new zmq.Dealer({ routingId }))
+            iopub: this.generateChannel(
```

Comment:
Maybe a comment about the highWatermark stuff so next person knows why this is here?